pkg/operation/botanist/component/istio/ingress_gateway.go

```diff
@@ -16,26 +16,18 @@ package istio
 
 import (
 	"context"
-	"path/filepath"
 
 	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
-	"github.com/gardener/gardener/pkg/client/kubernetes"
-	"github.com/gardener/gardener/pkg/operation/botanist/component"
-	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/gardener/gardener/pkg/chartrenderer"
 
-	networkingv1alpha3 "istio.io/client-go/pkg/apis/networking/v1alpha3"
-	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	crclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-type ingress struct {
-	values    *IngressValues
-	namespace string
-	kubernetes.ChartApplier
-	chartPath string
-	client    crclient.Client
+// IngressGateway is a set of configuration values for the istio-ingress chart.
+type IngressGateway struct {
+	Values    *IngressValues
```

Comment:
Why is this a pointer? I think we typically don't use pointers here.

Reply:
Pointer will make the program fast as it will not make a copy of variables. Is there a specific reason to not use pointer? Also, pointer was used in the previous code.

Reply:
I don't think that we have hit the point where we have to make such kind of optimizations. With this argument everything should be made a pointer.
In this case, for symmetry reasons in other components and because the values are not optional we should remove it.